source/wp-content/themes/wporg-developer-2023/src/style/style.scss

```diff
@@ -8,6 +8,12 @@ body[class] {
 	--wp-local-header-offset: 57px;
 }
 
+p,
```

Comment:
Can't find a way to do this through theme.json

Reply:
What is it you're trying to do? Would adjusting the block gap help?

The root block gap can be set in the `styles` property in theme.json:

```json
{
	"styles": {
		"spacing": {
			"blockGap": "var(--wp--preset--spacing--20)"
		}
	}
}
```

Reply:
Oh yeah maybe that's it. These elements are just using default browser margin at present

Reply:
No that doesn't seem to work

Reply:
What I'm trying to do is something like

```
"elements": {
			"p": {
				"spacing": {
					"blockGap": "var(--wp--preset--spacing--20)"
				}
			}
		},
```

But that isn't valid, the list of available elements is very limited according to the schema

Reply:
Hm, are you trying to add this spacing inside another block like the description?

<img width="726" alt="Screenshot 2023-08-30 at 6 27 25 PM" src="https://github.com/WordPress/wporg-developer/assets/541093/5647157a-58a4-4695-b97b-500e44faaa13">

I bet it's not working because it expects the custom block to handle it… it looks like you can trigger the block gap CSS by adding layout support to the description block (and others that need this), but that doesn't seem right, since you don't actually want full layout support.

```diff
	"supports": {
		"html": false,
		"spacing": {
			"margin": [ "top", "bottom" ],
			"padding": true
		},
		"typography": {
			"fontSize": true,
			"lineHeight": true
-		}
+		},
+		"layout": {
+			"default": {
+				"type": "constrained"
+			}
+		}
	},
```

Maybe mimicking the CSS in the block is a better idea? Like dropping this into `src/code-description/style.scss`:

```css
:where(.wp-block-wporg-code-reference-description) > * {
	margin-block-start: var(--wp--style--block-gap);
	margin-block-end: 0;
}

:where(.wp-block-wporg-code-reference-description) > :first-child {
	margin-block-start: 0;
}

:where(.wp-block-wporg-code-reference-description) > :last-child {
	margin-block-end: 0;
}
```

The `--wp--style--block-gap` variable should match whatever the theme has set as the default block gap, not the current block gap in the hierarchy of the page (noteworthy since this is inside the `post-content` block which sets blockGap to `var(--wp--preset--spacing--40)`).

Hopefully that helps a bit? 🙂

Reply:
Yeah thanks that is helpful 🙂 

I've added a [commit](https://github.com/WordPress/wporg-developer/pull/251/commits/5f5a6a620b167290d82fc42adb4c1b5ccd843032) which uses the `--wp--style--block-gap` everywhere I need it in this page. Seems like so much more code required than being able to set base element styles through the theme settings.

Let me know if you think that's the right approach